.github/workflows/CoverageCollection.yml

```diff
@@ -62,35 +72,76 @@ jobs:
       - name: 🗂 Install dependencies
         run: |
           python -m pip install -U pip
+          python -m pip install tomli
           python -m pip install ${{ inputs.requirements }}
 
+      - name: 🔁 Extract configurations from pyproject.toml
+        id: getVariables
+        shell: python
+        run: |
+          from pathlib import Path
+          from tomli   import load as tomli_load
+
+          htmlDirectory = 'htmlcov'
+          xmlFile = './coverage.xml'
+          coverageRC = "${{ inputs.coverage_config }}".strip()
+
+          # Read output paths from 'pyproject.toml' file
+          if coverageRC == "pyproject.toml":
+              pyProjectFile =  Path("pyproject.toml")
+              if pyProjectFile.exists():
+                  with pyProjectFile.open("rb") as file:
+                      pyProjectSettings = tomli_load(file)
+
+                  htmlDirectory = pyProjectSettings["tool"]["coverage"]["html"]["directory"]
+                  xmlFile       = pyProjectSettings["tool"]["coverage"]["xml"]["output"]
+              else:
+                  print(f"File '{pyProjectFile}' not found and no ' .coveragerc' file specified.")
+
+          # Read output paths from '.coveragerc' file
+          elif len(coverageRC) > 0:
+              coverageRCFile = Path(coverageRC)
+              if coverageRCFile.exists():
+                  with coverageRCFile.open("rb") as file:
+                      coverageRCSettings = tomli_load(file)
+
+                  htmlDirectory = coverageRCSettings["html"]["directory"]
+                  xmlFile       = coverageRCSettings["xml"]["output"]
+              else:
+                  print(f"File '{coverageRCFile}' not found.")
+
+          print(f"::set-output name=coverage_report_html_directory::{htmlDirectory}")
+          print(f"::set-output name=coverage_report_xml::{xmlFile}")
+          print(f"DEBUG:\n  html={htmlDirectory}\n  xml={xmlFile}")
+
       - name: Collect coverage
         continue-on-error: true
         run: |
-          python -m pytest -rA --cov=.. --cov-config=tests/.coveragerc tests/unit --color=yes
+          [ 'x${{ inputs.coverage_config }}' != 'x' ] && PYCOV_ARGS='--cov-config=${{ inputs.coverage_config }}' || unset PYCOV_ARGS
+          python -m pytest -rA --cov=. $PYCOV_ARGS ${{ inputs.unittest_directory }} --color=yes
 
       - name: Convert to cobertura format
         run: coverage xml
 
       - name: Convert to HTML format
         run: |
-          coverage html
-          rm htmlcov/.gitignore
+          coverage html -d ${{ steps.getVariables.outputs.coverage_report_html_directory }}
```

Comment:
Why do you specify the directory with `-d`?
This directory is taken from `pyproject.toml` oder `coveragerc` file.

Reply:
With `.coveragerc`, the report is written in `htmlcov`: https://github.com/edaa-org/pySVModel/runs/4633523272?check_suite_focus=true#step:8:8. It seems that the `html` field of the config file is not taken into account.

Reply:
We decided to support the three use cases for now, but we will recommend migrating to `pyproject.toml`. Other cases might be deprecated and removed in the future.